src/client/MealsComponents/MealDetail.js

```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import { useState, useEffect } from "react";
+import { Link } from "react-router-dom";
+import { MakeReservation } from "../ReservationComponents/Reservation";
+
+function MealsListDetails({ fetchMeal }) {
+  if (fetchMeal) {
+    const detailsOfMeals = fetchMeal.map((meal) => (
+      <li key={meal.id} className="mealdetailsSection">
+        <Link to={`/meals/${meal.id}`}>
+          <h3>{meal.title}</h3>
+        </Link>
+
+        <p>
+          Description : {meal.description}
+          <br></br>
+          Price : {meal.price} Kr
+          <br></br>
+          Location: {meal.location}
+          <br></br>
+          Max Reservation : {meal.max_reservations}
+          <br></br>
+          Meal_id : {meal.id}
+        </p>
+      </li>
+    ));
```

Comment:
```suggestion
    function createMealListItem(meal) {
      return (
      <li key={meal.id} className="mealdetailsSection">
        <Link to={`/meals/${meal.id}`}>
          <h3>{meal.title}</h3>
        </Link>

        <p>
          Description : {meal.description}
          <br></br>
          Price : {meal.price} Kr
          <br></br>
          Location: {meal.location}
          <br></br>
          Max Reservation : {meal.max_reservations}
          <br></br>
          Meal_id : {meal.id}
        </p>
      </li>
    );
    }
    
    const detailsOfMeals = fetchMeal.map(createMealItem);
    return (
      <ul className="mealList">
        {detailsOfMeals}
      </ul>
      );
```

I like to extract my mapping functions and give them names, it explains more about the code. You also see here that maybe this HTML should be it's own component!